presto-main/src/test/java/com/facebook/presto/sql/planner/assertions/PlanMatchPattern.java

```diff
@@ -59,6 +59,11 @@ public static PlanMatchPattern anyTree(PlanMatchPattern... sources)
         return any(sources).matchToAnyNodeTree();
     }
 
+    public static PlanMatchPattern anyNot(Class<? extends PlanNode> excludeNodeClass, PlanMatchPattern... sources)
```

Comment:
Talked offline with @martint. We're in agreement that reworking this is better done when the new optimizer goes in.